### ISSUE TITLE
Customrecipe doesn't save nbt/names of ingredients

### ISSUE THREAD

**Issue by Henriiig** [Zrips/CMI#8927]
### Description of Issue

When creating a custom recipe the custom item data is not saved for the ingredients so afterwards simply just the item is needed

![image](https://github.com/Zrips/CMI/assets/75689956/002dcf90-8ce3-4b2c-8f7d-7ba0d4931e4c)

### Version Information

```shell
[21:15:57 INFO]: --------------------------------------------------
[21:15:57 INFO]: CMI: 9.7.4.4 Velocity CMIB  SqLite
[21:15:57 INFO]: CMILib: 1.5.0.6 
[21:15:57 INFO]: Server: Spigot 1.8.8-R0.1-
[21:15:57 INFO]: CMI economy: Enabled Vault: 1.7.3-CMI CMI Chat: Enabled 
[21:15:57 INFO]: Modules -> 36 enabled 30 disabled: spawnerProximity, tablist, moneyCheque, skin, elytraBoost, elytraLaunch, bossBarCompass, chatBubble, playerCombat, votifier, shulkerBackpack, headDrop, spawnerCharge, vanish, namePlates, timedCommands, homeInteractions, signEdit, hpBossBar, launchPad, elytraExploit, anvilRenameColor, jail, spawners, kits, flightCharge, deathMessages, worldLimitsElytra, ranks, totemBehavior
[21:15:57 INFO]: --------------------------------------------------
```

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by Henriiig** [comment]
ALSO: It is not possible to create the same shapeless recipe using different types of dye because the game thinks that the recipe is already existing in 1.8 because dye is INK_SACK. Please fix this by checking for data of the item.

![image](https://github.com/Zrips/CMI/assets/75689956/34068984-b499-41fa-b2ac-f8545e5b98f3)

**Comment by Zrips** [comment]
Custom recipes uses original recipe handling, which can only recognize based on materials name only, aka original material name. Some of those only have one base material name with different sub data values, like colored wool or dyes. This is resolved and fixed with newer versions which can even recognize items by different model data information, but it's all based on how server recognizes it as entire handling is left for the server, CMI only creates recipes. 
In short, due to 1.8.8 version limitations, it's not possible to separate those items.
